src/daemon/bitcoind/actions.rs

```diff
@@ -587,16 +546,102 @@ pub fn bitcoind_main_loop(
+    let res = bitcoind.get_wallet_transaction(txid);
+    if let Ok((hex, blockheight, received_time)) = res {
+        Some(WalletTransaction {
+            hex,
+            blockheight,
+            received_time,
+        })
+    } else {
+        log::trace!(
+            "Got '{:?}' from bitcoind when requesting wallet transaction '{}'",
+            res,
+            txid
+        );
+        None
+    }
+}
+
+/// The bitcoind event loop.
+/// Listens for bitcoind requests (wallet / chain) and poll bitcoind every 30 seconds,
+/// updating our state accordingly.
+pub fn bitcoind_main_loop(
+    rx: Receiver<BitcoindMessageOut>,
+    revaultd: Arc<RwLock<RevaultD>>,
+    bitcoind: Arc<RwLock<BitcoinD>>,
+) -> Result<(), BitcoindError> {
+    // The verification progress announced by bitcoind *at startup* thus won't be updated
+    // after startup check. Should be *exactly* 1.0 when synced, but hey, floats so we are
+    // careful.
+    let sync_progress = Arc::new(RwLock::new(0.0f64));
+    // Used to shutdown the poller thread
+    let shutdown = Arc::new(AtomicBool::new(false));
+
+    // We use a thread to 1) wait for bitcoind to be synced 2) poll listunspent
+    let pol_revaultd = revaultd.clone();
+    let pol_bitcoind = bitcoind.clone();
+    let pol_sp = sync_progress.clone();
+    let pol_sd = shutdown.clone();
+    let poller_thread =
+        std::thread::spawn(move || poller_main(pol_revaultd, pol_bitcoind, pol_sp, pol_sd));
+
+    for msg in rx {
+        log::info!("Received {:?}", msg);
+        match msg {
+            BitcoindMessageOut::Shutdown => {
+                log::info!("Bitcoind received shutdown from main. Exiting.");
+                shutdown.store(true, Ordering::Relaxed);
+                assume_ok!(
+                    assume_ok!(poller_thread.join(), "Joining bitcoind poller thread"),
+                    "Error in bitcoind poller thread"
+                );
+                return Ok(());
+            }
+            BitcoindMessageOut::SyncProgress(resp_tx) => {
+                let s = assume_ok!(sync_progress.read(), "AAAAAAAAAAAAAA");
+                resp_tx.send(*s).map_err(|e| {
+                    BitcoindError::Custom(format!(
+                        "Sending synchronization progress to main thread: {}",
+                        e
+                    ))
+                })?;
+            }
+            BitcoindMessageOut::WalletTransaction(txid, resp_tx) => {
+                log::trace!("Received 'wallettransaction' from main thread");
+                resp_tx
+                    .send(wallet_transaction(&bitcoind.read().unwrap(), txid))
+                    .map_err(|e| {
+                        BitcoindError::Custom(format!(
+                            "Sending wallet transaction to main thread: {}",
+                            e
+                        ))
+                    })?;
+            }
+        }
     }
+
+    log::info!("Gamse over");
```

Comment:
`s/gamse/game`
```suggestion
    log::info!("Exiting");
```

Reply:
Lol that's some debugging leftover :sweat_smile: